In [19]:
import math
import numpy as np
import pandas as pd
from scipy.stats import norm

### Votaciones de hombres y mujeres en un partido determinado

Se obtuvieron los resultados de la votación en un pueblo cercano donde están registradas 48 mujeres y 40 hombres  
Será cierto que ¿Menos del 60% de las mujeres votaron por el partido que quedó en el poder?

In [8]:
votaciones = pd.DataFrame({
    'mujeres': {
        0: 1,
        1: 1,
        2: 1,
        3: 1,
        4: 1,
        5: 1,
        6: 1,
        7: 1,
        8: 0,
        9: 0,
        10: 1,
        11: 1,
        12: 1,
        13: 0,
        14: 0,
        15: 0,
        16: 1,
        17: 1,
        18: 1,
        19: 0,
        20: 1,
        21: 0,
        22: 1,
        23: 1,
        24: 1,
        25: 0,
        26: 0,
        27: 1,
        28: 1,
        29: 1,
        30: 0,
        31: 0,
        32: 1,
        33: 0,
        34: 0,
        35: 1,
        36: 1,
        37: 0,
        38: 0,
        39: 1,
        40: 1,
        41: 1,
        42: 0,
        43: 1,
        44: 1,
        45: 1,
        46: 1,
        47: 1
    },
    'hombres': {
        0: 0.0,
        1: 1.0,
        2: 1.0,
        3: 0.0,
        4: 1.0,
        5: 0.0,
        6: 0.0,
        7: 0.0,
        8: 1.0,
        9: 1.0,
        10: 0.0,
        11: 0.0,
        12: 0.0,
        13: 1.0,
        14: 1.0,
        15: 1.0,
        16: 0.0,
        17: 1.0,
        18: 0.0,
        19: 0.0,
        20: 1.0,
        21: 1.0,
        22: 1.0,
        23: 0.0,
        24: 0.0,
        25: 1.0,
        26: 1.0,
        27: 1.0,
        28: 0.0,
        29: 1.0,
        30: 0.0,
        31: 0.0,
        32: 1.0,
        33: 1.0,
        34: 0.0,
        35: 1.0,
        36: 0.0,
        37: 1.0,
        38: 0.0,
        39: 0.0,
        40: np.nan,
        41: np.nan,
        42: np.nan,
        43: np.nan,
        44: np.nan,
        45: np.nan,
        46: np.nan,
        47: np.nan
    }
})

In [14]:
votaciones_mujeres = votaciones["mujeres"][votaciones["mujeres"].notnull()]

### 1. Datos
Los datos provienen de los registros de votantes de una comunidad. Nos interesa estudiar la proporción de mujeres que votaron por el partido que quedó en el poder. Nivel de significancia del 0.05.
### 2. Supuestos  
Las respuestas de las mujeres se modelan según una variable aleatoria del tipo Bernoulli, donde no conocemos la varianza (pero se puede estimar) y que tenemos una muestra suficientemente grande $n=48$ para $n>30$
### 3. Hipótesis
* Hipótesis de Investigación: La proporción de mujeres votantes es menor al 60%
* $H_a$: $p < 0.6$  
* $H_0$: $p \geq 0.6$  
Planteamos que $H_0$ es verdadera y queremos demostrar que es falsa
### 4. Estadístico de prueba
$Z = \frac{\bar{p}-p_0}{\sqrt{\frac{p_0*q_0}{n}}}$
### 5. Distribución del estadístico de prueba
$Z$ es $N(0,1)$ cuando $n \rightarrow \infty$ utilizando TCL
### 6. Región de rechazo
Al ser unilateral
* $V.C. = - Z_{0.95}$

In [17]:
alpha = 0.05
p0 = 0.6
q0 = 1 - p0

In [13]:
# Valor crítico
vc_superior = norm.ppf(alpha)
vc_superior

-1.6448536269514729

Región de Rechazo $= \{\infty, -1.6448536\} $

### 7. Cálculo del valor del estadístico de prueba

In [16]:
p_est = votaciones_mujeres.mean()
p_est

0.6666666666666666

In [18]:
n  = len(votaciones_mujeres)
n

48

In [22]:
z = (p_est - p0) / math.sqrt(p0 * q0 / n)
z

0.9428090415820631

### 8. Decisión  
Debido a que el valor del estadístico de prueba $(z)$ del paso anterior no se encuentra en la región de rechazo (6), por lo tanto la hipótesis nula no es rechazada

### 9. Conclusión
Después de realizar el estudio pertinente encontramos que la proporción de mujeres que votaron por el partido en el poder quizá es mayor al 60% que se propuso  
### 10. P-valor

In [25]:
p_value = norm.cdf(z)
p_value

0.8271107069244198

Como $p_{value} > \alpha = 0.05$ (la muestra es significativa); Donde $H_0$ no se rechazó